spec/requests/alert_definitions_spec.rb

```diff
@@ -306,4 +306,46 @@
     expect(alert_definition_profiles.first.reload.description).to eq("Updated Test Alert Profile 1")
     expect(alert_definition_profiles.second.reload.description).to eq("Updated Test Alert Profile 2")
   end
+
+  it "assings an alert definition to a profile" do
+    api_basic_authorize collection_action_identifier(:alert_definition_profiles, :edit)
+    alert_definition_profile = FactoryGirl.create(:miq_alert_set)
+    alert_definition = FactoryGirl.create(:miq_alert)
+
+    post(api_alert_definition_profile_alert_definitions_url(nil, alert_definition_profile), :params => gen_request(:assign, alert_definition))
+
+    expect(response).to have_http_status(:ok)
+    expect(response.parsed_body["results"].count).to eq(1)
+    expect_result_resources_to_include_hrefs(
+      "results",
+      [api_alert_definition_profile_alert_definition_url(nil, alert_definition_profile, alert_definition)]
+    )
+  end
+
+  it "unassings an alert definition from a profile" do
+    api_basic_authorize collection_action_identifier(:alert_definition_profiles, :edit)
+    alert_definition_profile = FactoryGirl.create(:miq_alert_set)
+    alert_definition = FactoryGirl.create(:miq_alert)
+    alert_definition_profile.add_member(alert_definition)
+
+    post(api_alert_definition_profile_alert_definitions_url(nil, alert_definition_profile), :params => gen_request(:unassign, alert_definition))
+
+    expect(response).to have_http_status(:ok)
+    expect(response.parsed_body["results"].count).to eq(1)
+    expect_result_resources_to_include_hrefs(
+      "results",
+      [api_alert_definition_profile_alert_definition_url(nil, alert_definition_profile, alert_definition)]
+    )
+  end
+
+  it "unassings an alert definition from a profile to which it wasn't assigned" do
+    api_basic_authorize collection_action_identifier(:alert_definition_profiles, :edit)
+    alert_definition_profile = FactoryGirl.create(:miq_alert_set)
+    alert_definition = FactoryGirl.create(:miq_alert)
+
+    post(api_alert_definition_profile_alert_definitions_url(nil, alert_definition_profile), :params => gen_request(:unassign, alert_definition))
+
+    expect(response).to have_http_status(:ok)
```

Comment:
This looks OK, but can you check whether this is consistent with unassigning behavior in other APIs (tags, for instance).

Reply:
assign, unassign actions are fine, consistent with tagging, policies, etc.  but the responses should use
the action result signature, i.e. ('success' : ..., 'message' : ..., 'href' : ..., 'alert_definition_href' : ....,)

Reply:
if these can have multiple alert_definition_profile -> alert_definition associations, bulk request tests are also needed.